File: src/com/github/jrd77/codecheck/data/MatchRule.java
package com.github.jrd77.codecheck.data;

/**
 * @author zhen.wang
 * @description TODO
 * @date 2021/8/19 14:44
 */
public class MatchRule {

    private String rule;
    private RuleTypeEnum ruleType;
    private String comment;


    public MatchRule(String rule, RuleTypeEnum ruleType, String comment) {
        this.rule = rule;
        this.ruleType = ruleType;
        this.comment = comment;
    }

    public MatchRule() {
    }

    @Override
    public String toString() {
        return "MatchRule{" +
                "rule='" + rule + '\'' +
                ", ruleType='" + ruleType + '\'' +
                ", comment='" + comment + '\'' +
                '}';
    }

    public String getRule() {
        return rule;
    }

    public void setRule(String rule) {
        this.rule = rule;
    }

    public RuleTypeEnum getRuleType() {
        return ruleType;
    }

    public void setRuleType(RuleTypeEnum ruleType) {
        this.ruleType = ruleType;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }
}

File: src/com/github/jrd77/codecheck/data/MatchRule.kt
package com.github.jrd77.codecheck.data

import com.github.jrd77.codecheck.data.RuleTypeEnum
import java.util.*

/**
 * @author WZ
 */
class MatchRule {
    var rule: String? = null
    var ruleType: RuleTypeEnum? = null
    var comment: String? = null

    constructor() {}
    constructor(rule: String?, ruleType: RuleTypeEnum?, comment: String?) {
        this.rule = rule
        this.ruleType = ruleType
        this.comment = comment
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val matchRule = o as MatchRule
        return rule == matchRule.rule &&
                ruleType === matchRule.ruleType
    }

    override fun hashCode(): Int {
        return Objects.hash(rule, ruleType)
    }

    override fun toString(): String {
        return "MatchRule{" +
                "rule='" + rule + '\'' +
                ", ruleType=" + ruleType +
                ", comment='" + comment + '\'' +
                '}'
    }
}